src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,308 @@
+import React, { useEffect } from 'react';
+import classNames from 'classnames';
+import { NavLink, useParams } from 'react-router-dom';
+import { client } from '../../utils/fetchClient';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { PeopleFilters } from '../PeopleFilters';
+import { useTableContext } from '../Context';
+import { SearchLink } from '../SearchLink/SearchLink';
+
+export const PeoplePage: React.FC = () => {
+  const {
+    peoples,
+    setPeoples,
+    hasError,
+    setHasError,
+    isLoading,
+    setIsLoading,
+    currPeoples,
+    searchParams,
+    query,
+    gender,
+    centuries,
+    setCurrPeoples,
+    sortCurr,
+    order,
+  } = useTableContext();
+
+  const necessaryPeople = currPeoples ?? peoples;
+
+  const { slug } = useParams();
+
+  const sortParams = (sortType: string) => {
+    if (sortCurr !== sortType) {
+      return { sort: sortType, order: null };
+    }
+
+    if (order) {
+      return { sort: null, order: null };
+    }
+
+    return { sort: sortType, order: 'desc' };
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    client.get('people.json')
+      .then((response: unknown) => {
+        setPeoples(response as Person[]);
+      })
+      .catch(() => {
+        setHasError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    let filteredPeople = peoples;
+
+    if (query !== '') {
+      filteredPeople = filteredPeople.filter(person => {
+        const normalizeQuery = query.toLowerCase();
+        const normalizeName = person.name.toLowerCase();
+
+        return (
+          normalizeName.includes(normalizeQuery)
+          || (person.fatherName && person.fatherName
+            .toLowerCase().includes(normalizeQuery))
+          || (person.motherName && person.motherName
+            .toLowerCase().includes(normalizeQuery))
+        );
+      });
+    }
+
+    if (gender !== '') {
+      filteredPeople = filteredPeople.filter(person => person.sex === gender);
+    }
+
+    if (centuries.length > 0) {
+      filteredPeople = filteredPeople
+        .filter(person => {
+          const century = Math.ceil(person.died / 100);
+
+          if (centuries.includes(century.toString())) {
+            return person;
+          }
+
+          return null;
+        });
+    }
+
+    if (sortCurr !== '') {
+      filteredPeople = filteredPeople.sort((a, b) => {
+        switch (sortCurr) {
+          case 'sex':
+          case 'name':
+            return a[sortCurr].localeCompare(b[sortCurr]);
+
+          case 'born':
+          case 'died':
+            return a[sortCurr] - b[sortCurr];
+
+          default: return 0;
+        }
+      });
+
+      if (order) {
+        filteredPeople = filteredPeople.reverse();
+      }
+
+      setCurrPeoples(filteredPeople);
+    }
+
+    setCurrPeoples(filteredPeople);
+  }, [peoples, searchParams]);
```

Comment:
I guess your problem is here, because you filter your people array after search params change but consider to do this after click on filter buttons in the head of table